clang/include/clang/Basic/Attr.td

```diff
@@ -2031,6 +2031,13 @@ def AMDGPUNumVGPR : InheritableAttr {
   let Subjects = SubjectList<[Function], ErrorDiag, "kernel functions">;
 }
 
+def AMDGPUNumWorkGroups : InheritableAttr {
+  let Spellings = [Clang<"amdgpu_num_work_groups", 0>];
+  let Args = [UnsignedArgument<"NumWorkGroupsX">, UnsignedArgument<"NumWorkGroupsY">, UnsignedArgument<"NumWorkGroupsZ">];
```

Comment:
`UnsignedArgument` doesn't allow dependent values.  Is this something we're OK with?  Typically we'd want to support some level of template support, which means you have to store as expressions and convert when needed (then TreeTransform it properly).

Reply:
Changed `UnsignedArgument` to `AttrArgument`.